vowpalwabbit/bs.h

```diff
@@ -11,9 +11,9 @@ LEARNER::base_learner* bs_setup(VW::config::options_i& options, vw& all);
 
 namespace BS
 {
-inline uint32_t weight_gen(vw& all)  // sampling from Poisson with rate 1
+inline uint32_t weight_gen(rand_state & state)  // sampling from Poisson with rate 1
 {
-  float temp = merand48(all.random_state);
+  float temp = state.get_and_update_random();
```

Comment:
Would it make sense to just pass in the state(float) instead of rand_state?

Reply:
It seems like this function was intended to work specifically off the random state and advance it. I'm not sure if it should be generalized beyond that